my_enums.rb

```diff
@@ -1,60 +1,181 @@
 # frozen_string_literal: true
 
-# rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity
+# rubocop: disable Metrics/AbcSize
+# rubocop: disable Metrics/CyclomaticComplexity
+# rubocop: disable Metrics/MethodLength
+# rubocop: disable Metrics/PerceivedComplexity
 
 module Enumerable #:nodoc: all
   def my_each
+    return to_enum unless block_given?
+
     length.times do |n|
       current = self[n]
       yield(current)
     end
+    self
   end
 
   def my_each_with_index
-    length.times do |n|
-      index = n
-      current = self[n]
-      yield(index, current)
+    return to_enum unless block_given?
+
+    my_each do |n|
+      yield(n, index(n))
     end
+    self
   end
 
   def my_select
     emp_arr = []
+    return to_enum unless block_given?
+
     my_each do |n|
       emp_arr << n if yield(n)
     end
     emp_arr
   end
 
-  def my_all?
-    emp_arr = []
-    my_each do |n|
-      emp_arr << n if yield(n)
+  def my_all?(param = nil)
+    if block_given?
+      begin
+        if yield.is_a? Class
+          my_each do |n|
+            return false unless n.is_a? yield
+          end
+          true
+        elsif yield.instance_of? Regexp
+          my_each do |n|
+            next if n =~ yield
+
+            return false
+          end
+          true
+        end
+      rescue NoMethodError
+        my_each do |n|
+          next if yield(n)
+
+          return false
+        end
+        true
+      end
+    elsif !param.nil?
+      if param.is_a? Class
+        my_each do |n|
+          return false unless n.is_a? param
+        end
+      elsif param.instance_of? Regexp
+        my_each do |n|
+          next if n =~ param
+
+          return false
+        end
+      else
+        my_each do |n|
+          next if n == param
+
+          return false
+        end
+      end
+      true
+    else
+      return false if empty?
+
+      my_each do |n|
+        return false if n.nil? || n == false
+      end
+      true
     end
-    emp_arr == self
   end
 
   def my_any?
-    tester = false
-    my_each do |n|
-      tester = true if yield(n)
-      break if tester
+    if block_given?
+      begin
+        if yield.is_a? Class
+          my_each do |n|
+            return true if n.is_a? yield
+          end
+          false
+        elsif yield.instance_of? Regexp
+          my_each do |n|
+            return true if n =~ yield
+          end
+          false
+        end
+      rescue NoMethodError
+        my_each do |n|
+          return true if yield(n)
+        end
+        false
+      end
+    else
+      return false if empty?
+
+      my_each do |n|
+        return true unless n.nil? || n == false
+      end
+      false
     end
```

Comment:
* [ ] enumerables my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class
```
    array = [1,2,5,'4',7]
    p array.my_any?(Integer) == array.any?(Integer) # false
```

* [ ] enumerables my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
  ```
    array = %w[dog door rod blade]
    array[1] = 9
    p array.my_any?(/d/) == array.any?(/d/) # false
  ```

* [ ] enumerables my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
  ```
    array = %w[dog door rod blade]
    p array.my_any?('dog') == array.any?('dog') # false
  ```